mythril/laser/ethereum/instructions.py

```diff
@@ -619,23 +619,24 @@ def signextend_(self, global_state: GlobalState) -> List[GlobalState]:
         :param global_state:
         :return:
         """
-        state = global_state.mstate
-        s0, s1 = state.stack.pop(), state.stack.pop()
+        mstate = global_state.mstate
+        s0, s1 = mstate.stack.pop(), mstate.stack.pop()
 
         try:
             s0 = util.get_concrete_int(s0)
             s1 = util.get_concrete_int(s1)
         except TypeError:
-            return []
+            mstate.stack.append(s1)
```

Comment:
Adding a TODO here about something more appropriate might be good.

Reply:
And how about pushing a new variable named "SIGNEXTEND(hash(s0), hash(s1))"? This also has it's own problems.